### ISSUE TITLE
Active section not set correctly if section has no content

### ISSUE THREAD

**Issue by asprouse5** [tscanlin/tocbot#253]
I noticed that if a section has no or very little content the active section isn't set correctly. See https://jsfiddle.net/1ctge5my/1/, also summarized below. In the fiddle the section header appears behind the fixed header but I'm not seeing that issue on my site.

To reproduce: Click on "Three", then click on "One". I would expect "One" to be the active section, but it's actually "Two". I suspect the headingsOffset has something to do with it, but I think I need to use that for my fixed header. Uncomment the paragraph in the first section and it works as expected.

```html
<div id="header"></div>
<div>
  <div id="box">
    <div id="sections">
      <section>
        <header>
          <h2 id="One">One</h2>
        </header>
        <!-- <p>
          One Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p> -->
      </section>
      <section>
        <header>
          <h2 id="Two">Two</h2>
        </header>
        <p>
          Two Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </section>
      <section>
        <header>
          <h2 id="Three">Three</h2>
        </header>
        <p>
          Three Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </section>
    </div>
    <div id="toc"></div>
  </div>
</div>
```
```css
body {
  min-height: 1000px;
  position: absolute;
  top: 50px;
  bottom: 0;
  overflow: auto;
}

#header {
  position: fixed;
  top: 0;
  height: 50px;
  width: 100%;
  background-color: blue;
}

#box {
  display: flex;
}

#toc {
  position: sticky;
  top: 60px;
  align-self: flex-start;
}

.toc-list {
  list-style: none;
}
```

```js
tocbot.init({
  tocSelector: '#toc',
  contentSelector: '#sections',
  headingSelector: 'h2',
  headingsOffset: 50,
  scrollSmooth: false
});
```

**Comment by tscanlin** [comment]
Hmm, when I try that fiddle you sent it seems to scroll fine for me.
What OS / browser are you using? Can you include a screenshot?

**Comment by asprouse5** [comment]
I've tried on a Mac using Firefox and Chrome, and in Windows using Firefox and Edge. This is the state of the screen when I first click "Three", then click "One".

<img width="704" alt="Screen Shot 2021-11-22 at 9 40 28 AM" src="https://user-images.githubusercontent.com/39681760/142880909-a9f3bb6d-bb1d-4e55-b1b2-46f443919db6.png">

Another issue came up as well. I occasionally see an error when running unit tests, and I think it's because activeTocLink can be null but you don't check for it in every case. The code is in build-html.js:
```js
      // Add the active class to the active tocLink.
      var activeTocLink = document.querySelector(options.tocSelector)
        .querySelector('.' + options.linkClass +
          '.node-name--' + topHeader.nodeName +
          '[href="' + options.basePath + '#' + topHeader.id.replace(/([ #;&,.+*~':"!^$[\]()=>|/@])/g, '\\$1') + '"]')
      if (activeTocLink && activeTocLink.className.indexOf(options.activeLinkClass) === -1) {
        activeTocLink.className += SPACE_CHAR + options.activeLinkClass
      }
      var li = activeTocLink.parentNode        // this is the line that throws the error
```
And the error states `Uncaught TypeError: Cannot read properties of null (reading 'parentNode') thrown`. I'm having to disable the tests that call tocbot for now, but would like to be able to test them in the future.

**Comment by tscanlin** [comment]
Hey, I can add an extra check there to prevent that error around `activeTocLink.parentNode. I've not had other people run into this before though, are the ID's you are using special in any way? Also, does the prior issue still happen if you have content or does adding a paragraph below remedy the issue?

**Comment by asprouse5** [comment]
I don't believe the ids are special. They are mostly alphabet characters and might have words in parentheses. I forced all the ids to be all alphabet characters and the issue remained. A null check would be great.

And no, the prior issue doesn't happen if you add content. Adding a paragraph below remedies the issue, so I think that will be the solution for us.

**Comment by tscanlin** [comment]
Thanks for the clarification. I've put up this PR which should address the issue:
https://github.com/tscanlin/tocbot/pull/256

Thanks for your help with this!

**Comment by tscanlin** [comment]
This is published in `4.17.1` feel free to ping back here if anything else comes up. Thank you!
